internal/jsonext/jsonext.go

```diff
@@ -0,0 +1,29 @@
+// Package jsonext defines common types with JSON marshalers.
+package jsonext
```

Comment:
@ashmrtn Not sure about the package name. Do you have a better suggestion?
If not, it will be easy to change later anyway.

Reply:
ah, I'm really not too great at naming 😅 About the only other thing I can think of is `jsonmarshal`, but it's not really much more descriptive than what's already in this PR

Reply:
@ashmrtn `jsonmarshal` works, what about `jsonencoding` (it's a little long though), thoughts?

The stdlib's package is named `encoding/json`.

Reply:
`jsonencoding` is fine with me